packages/web-config-server/src/apiV1/utils/findAccessibleGroupedMapOverlays.js

```diff
@@ -159,17 +159,29 @@ const sortMapOverlayItems = (mapOverlayItems, relations) => {
 const translateOverlaysForResponse = mapOverlays =>
   mapOverlays
     .filter(({ presentationOptions: { hideFromMenu } }) => !hideFromMenu)
-    .map(({ id, name, linkedMeasures, presentationOptions }) => {
-      const idString = [id, ...(linkedMeasures || [])].sort().join(',');
-
-      return {
-        id, // just for sorting purpose, will be removed later
-        measureId: idString,
-        code: idString,
+    .map(
+      ({
+        id,
         name,
-        ...presentationOptions,
-      };
-    });
+        linkedMeasures,
+        presentationOptions,
+        report_code: reportCode,
+        dataElementCode,
+        legacy,
+      }) => {
+        const idString = [id, ...(linkedMeasures || [])].sort().join(',');
+
+        return {
+          id, // just for sorting purpose, will be removed later
+          measureId: idString,
+          code: legacy ? idString : reportCode, // if we're not a legacy overlay, return the report code
```

Comment:
Hmmm...with the dashboard items, do we return the report code or the dashboard item code under the `code` key? It seems a little misleading here

Reply:
Ah fair enough. Dashboard items have a separate field for reportCode so I've updated this to match